### PR TITLE
Fix test failure in `test_spatial_transformer_sampler.py`

### LINKED ISSUE #6905: `test_spatial_transformer_sampler` is flaky
This is part of https://github.com/chainer/chainer/issues/6903 effort.

- Failure example: https://ci.preferred.jp/chainer.py27and35/2070/
- Failure test: chainer_tests/functions_tests/array_tests/test_spatial_transformer_sampler.py
- Failure test case: `TestSpatialTransformerSamplerConsistencyWithCuDNN_param_0_{dtype=float16}.test_consistency_with_cudnn_cpu`
- Tolerance:
```
E           Not equal to tolerance rtol=0.0001, atol=0.01
E           
E           Mismatch: 2.78%
E           Max absolute difference: 1.692
E           Max relative difference: nan
```

### PR DESCRIPTION
**Pull request by asi1024**
Fix #6905.

`SpatialTransformerSampler.backward` is discontinuous, so the current test failure cannot be fixed by only relaxing the tolerances.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/array_tests/test_spatial_transformer_sampler.py [hunk 1]
```diff
@@ -122,6 +122,7 @@ def _apply_backward(self, x, grid, grads):
         y.backward()
         return x, grid, y
 
+    @condition.repeat_with_success_at_least(100, 95)
```
Comment (asi1024) [inline tests/chainer_tests/functions_tests/array_tests/test_spatial_transformer_sampler.py]: This condition should be added only when `self.dtype == 'float16'`.

**Comment by niboshi** [comment]
How about using `testing.fix_random`?

**Comment by asi1024** [comment]
How can I make inputs deterministic only in float16 test?

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by niboshi** [comment]
Jenkins, test this please

**Comment by hvy** [comment]
LGTM!

**Pull request merged**